apps/aesophia/test/contracts/erc20_token.aes

```diff
@@ -0,0 +1,97 @@
+contract ERC20Token =
+  record state = {
+    totalSupply : int,
+    decimals : int,
+    name : string,
+    symbol : string,
+    balances : map(address, int),
+    allowed : map(address, map(address,int)),
+    // Logs, remove when native Events are there
+    transfer_log : list((address,address,int)),
+    approval_log : list((address,address,int))}
+
+  // init(100000000, 10, "Token Name", "TKN")
+  public stateful function init(_totalSupply : int, _decimals : int, _name : string, _symbol : string ) = {
+    totalSupply = _totalSupply,
+    decimals = _decimals,
+    name = _name,
+    symbol = _symbol,
+    balances = {[Call.caller] = _totalSupply }, // creator gets all Tokens
+    allowed = {},
+    // Logs, remove when native Events are there
```

Comment:
Could you please confirm which Pivotal Tracker ticket tracks this? If there is none, could you please add one and post the link here?

Reply:
I have added a link the ticket in the first comment.

Reply:
Nono - I meant which ticket tracks this (implicitly) TODO comment "Logs, remove when native Events are there". They are currently kept in memory. Once we have events in Sophia we need to remember to change this contract. A ticket in Pivotal specifically for this would enable us to remember.

Reply:
Tracked https://www.pivotaltracker.com/story/show/160750737